src/emqx_trie.erl

```diff
@@ -88,8 +88,17 @@ insert(Topic) when is_binary(Topic) ->
         [TrieNode = #trie_node{topic = undefined}] ->
             write_trie_node(TrieNode#trie_node{topic = Topic});
         [] ->
+            Paths = triples(Topic),
+            case length(ekka_mnesia:running_nodes()) > 1 andalso length(Paths) > 4 of
```

Comment:
this magic number 4 deserves some documentation.
even better if we can make it configurable.